src/bootstrap/src/core/build_steps/setup.rs

```diff
@@ -607,28 +674,32 @@ fn create_vscode_settings_maybe(config: &Config) -> io::Result<bool> {
         }
     };
     if should_create {
-        let path = config.src.join(".vscode");
+        let path = settings_path.as_path().parent().unwrap();
```

Comment:
Suggestion: we probably should have helpers that derive a set of `Path`s instead of doing parent navigations:

1. The `src/etc/` path: `config.src.join("etc")`.
    -  The template configs for each editor kind based on that: `template_editor_config(kind: EditorKind) -> PathBuf`. e.g. `src/etc/vscode-settings.json`.
2. User root editor settings folder that may or may not exist: `user_editor_settings_folder(kind: EditorKind) -> PathBuf)` => `config.src.join(".vscode")` or helix or emacs.
    - Based on that, user editor settings file, i.e. `config.src.join(".vscode/settings.json")`.

(I keep forgetting which `src` `config.src` refers to, I think it's the source checkout root, in which case `config.src.join("etc")` should be `config.src.join("src/etc")`).

Reply:
Good point, I refactored 2) into `editor.settings_folder()`, no more parent navigation.
1) is not really necessary I think, we only use the template path once and it's to read it into a str. From there setup only operates on the contents. All the path navigation works around where the actual config will go to.